sessiontxn/isolation/repeatable_read.go

```diff
@@ -179,24 +182,37 @@ func (p *PessimisticRRTxnContextProvider) AdviseOptimizeWithPlan(val interface{}
 		plan = execute.Plan
 	}
 
-	mayOptimizeForPointGet := false
-	if v, ok := plan.(*plannercore.PhysicalLock); ok {
-		if _, ok := v.Children()[0].(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
-		}
-	} else if v, ok := plan.(*plannercore.Update); ok {
-		if _, ok := v.SelectPlan.(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
+	p.optimizeForNotFetchingLatestTS = optimizeForNotFetchingLatestTS(plan)
+
+	return nil
+}
+
+// optimizeForNotFetchingLatestTS searches for optimization condition recursively
+func optimizeForNotFetchingLatestTS(plan plannercore.Plan) bool {
+	switch v := plan.(type) {
+	case *plannercore.PointGetPlan:
+		return true
```

Comment:
Can we add a additional check for `PointGetPlan/BatchPointGetPlan` that when they are in `SelectLock/Insert/Update` and `plan.lock` is false, this method should still return false

Reply:
I have  modified it.